src/libcollections/btree/node.rs

```diff
@@ -1063,6 +1067,40 @@ impl<'a, K, V> Handle<NodeRef<marker::Mut<'a>, K, V, marker::Internal>, marker::
             Handle::new_edge(self.node, self.idx)
         }
     }
+
+    pub fn steal_left(mut self)
```

Comment:
Well, yes, probably. `merge` is slightly different in that there isn't a way to append two nodes without it, but it should really be split into a `remove` followed by some sort of `Node::append` method that takes a `Root` of the same height. I guess I didn't completely follow my own rule.